simple-kotlin-kotlin-dsl/buildSrc/src/main/kotlin/provider.simple.kotlin.kotlin-application-conventions.gradle.kts

```diff
@@ -11,3 +14,53 @@ plugins {
     // Apply the application plugin to add support for building a CLI application in Java.
     application
 }
+
+operator fun <T> Property<T>.plusAssign(value: T) {
+    set(value)
+}
+operator fun <T> Property<T>.plusAssign(value: Provider<T>) {
+    set(value)
+}
+// Since we add the Property<String>.plusAssign(Any) override, we also need to re-add the other ones.
+// If we don't do that, then the Any override is used even if the target is a Provider<String>
+operator fun Property<String>.plusAssign(value: String) {
+    set(value)
+}
+// We need to add the JvmName annotation here, so the compilation doesn't fail with "Platform declaration clash"
```

Comment:
Is this because we have the same method signature because of generics erasure?

Reply:
Yes, for Java we'd define the same method because of type erasure.